src/components/AddressBookModal/AddressBookModal.vue

```diff
@@ -0,0 +1,217 @@
+          <div class="address-inputs">
+            <blockie
+              v-show="isValidAddress"
+              :address="contactAddress"
+              width="32px"
+              height="32px"
+              class="blockie-image"
+            />
+            <input
+              v-model="contactAddress"
+              v-addr-resolver="'contactAddress'"
+              :class="isValidAddress ? 'blockie-input' : ''"
+              :placeholder="$t('common.addr')"
+              type="text"
+            />
+          </div>
+        </div>
+        <div v-show="title === 'Edit Address'" class="edit-address">
+          <div class="addr-container">
+            <blockie
+              :address="contactAddress"
+              width="34px"
+              height="34px"
+              class="blockie-image mr-3"
+            />
+            <div>
+              <p class="title-header">
+                {{ $t('interface.address-book.address') }}
+              </p>
+              <p class="address">{{ contactAddress }}</p>
+            </div>
+          </div>
+        </div>
+        <div class="nickname-container">
+          <p class="title-header">
+            {{ $t('interface.address-book.nickname') }}
+          </p>
+          <div
+            :class="title === 'Edit Address' ? 'nickname-input-container' : ''"
+          >
+            <input
+              v-model="contactNickname"
+              :placeholder="$t('interface.address-book.nickname')"
+              class="nickname-input mt-3 mb-4"
+              type="text"
+            />
+            <div class="button-container">
+              <button
+                :class="isBtnDisabled ? 'disabled' : ''"
+                @click="updateAddrBook()"
+              >
+                {{
+                  title === 'Edit Address'
+                    ? $t('interface.address-book.update')
+                    : $t('interface.address-book.add')
+                }}
+              </button>
+            </div>
+          </div>
+        </div>
+        <p
+          v-show="title === 'Edit Address'"
+          class="remove-txt"
+          @click="removeContact()"
+        >
+          {{ $t('interface.address-book.remove-addr') }}
+        </p>
+      </div>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+import { Toast } from '@/helpers';
+import { mapState } from 'vuex';
+import Blockie from '@/components/Blockie';
+
+export default {
+  components: {
+    blockie: Blockie
+  },
+  props: {
+    title: {
+      type: String,
+      default: ''
+    },
+    currentIdx: {
+      type: Number,
+      default: 0
+    },
+    selectedAddress: {
+      type: String,
+      default: ''
+    }
+  },
+  data() {
+    return {
+      contactNickname: '',
+      contactAddress: '',
+      isValidAddress: false
+    };
+  },
+  computed: {
+    ...mapState(['addressBook']),
+    isBtnDisabled() {
+      if (this.title === 'Edit Address' && this.addressBook[this.currentIdx]) {
```

Comment:
Don't check against strings, export an enum
```
const seasons = {
    SUMMER: 'summer',
    WINTER: 'winter',
    SPRING: 'spring',
    AUTUMN: 'autumn'
}

season === seasons.SPRING
```
Then you can use it on both here and wherever you pass the prop

Reply:
👍 👍